inc/class-core-sitemaps.php

```diff
@@ -96,6 +97,13 @@ public function setup_sitemaps() {
 		}
 	}
 
+	/**
+	 * Provide rewrite for the xsl stylesheet.
+	 */
+	public function xsl_stylesheet_rewrite() {
```

Comment:
Ah, I see what's happening here.

The rewrite rule will need to direct to something like `index.php?sitemap-xsl=1`, and then you can add a method to the `Core_Sitemaps` class that works similar to how `Core_Sitemaps_Index::render_sitemap()` works, where you hook to the `template_redirect` action with a function that looks for the existence of the `sitemap-xsl` param—which you'll need to add an `add_rewrite_tag` for—and then include your sitemap.xsl.php file programmatically and `exit`.